### PR TITLE
Drones can now cancel essence links properly, essence link automatically removed on death.

### LINKED ISSUE #11332: Drone link continues to beam after a drone perishes
### Description

If a drooner is linked to you with the new drone link medi-beam thing and they die, the beam will continue to draw to you from their corpse, even if you move away and run back. I'm not sure if it applies any healing, but the beam still draws and it draws attention to you despite your pocket drone being a lifeless corpse.

I didn't think to take a screenshot because the round was super laggy when I noticed this happening and I didn't want to throw and die trying to take a picture for a bug report. Not sure if it makes a difference, but I was a praetorian at the time.

### Test Merges

Server-Side Code Modifications: https://tgstation13.org/parsed-logs/sybil/server_side_modifications.dm

Round ID: [17410](https://scrubby.melonmesa.com/round/17410)
BYOND Version: 514.1585
Server revision compiled on: 
Master commit: [af73cb5a187ee4479e42fb50efc36e8fbcfdfc30](https://github.com/tgstation/TerraGov-Marine-Corps//commit/af73cb5a187ee4479e42fb50efc36e8fbcfdfc30)
The following pull requests are currently test merged:
[#11189: 'makes mounted guns easy to use' by tyeagg at commit 6804324b36](https://github.com/tgstation/TerraGov-Marine-Corps//pull/11189)
[#11193: 'Drone Redesigned' by lKiseki at commit a3215800ee](https://github.com/tgstation/TerraGov-Marine-Corps//pull/11193)
[#11214: 'projectile based artillery + general stat rework' by tyeagg at commit d7d8b81fe5](https://github.com/tgstation/TerraGov-Marine-Corps//pull/11214)
[#11290: 'SADAR Changes, added 'Unguided' HE/WP rocket types, Reworked AP.' by Pariah919 at commit 74857997a3](https://github.com/tgstation/TerraGov-Marine-Corps//pull/11290)

Local commit: 2592483a876e876dcb90a7aa0c54b0fa595184c4
Server tools version: 5.2.2

### Reproduction Steps

1. Be beno
2. Enslave drooner to drone link to you
3. Drooner dies
4. Their beam still links to you from the corpse

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by Penwin0**
## About The Pull Request
`enhancement_action.end_ability()` was causing a runtime as drones will always lack that ability unless primordial.
Replaced `remove_status_effect` with `QDEL_NULL` accomplishes the same thing but faster, no reason not to use it as we are already storing a reference to it in the action.

closes #11332

## Why It's Good For The Game
Things should work as they are supposed to, drone link not being removed on death was also causing issue

## Changelog
:cl:
fix: Drone link is now removed on death
fix: Drone link removal by alt-use of the link ability now works
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm [hunk 1]
```diff
@@ -73,9 +73,10 @@
 /datum/action/xeno_action/activable/essence_link/proc/end_ability()
 	var/mob/living/carbon/xenomorph/X = owner
 	var/datum/action/xeno_action/enhancement/enhancement_action = X.actions_by_path[/datum/action/xeno_action/enhancement]
-	enhancement_action.end_ability()
+	if(enhancement_action)
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/drone/abilities_drone.dm:76]: use ?. instead

File: code/modules/mob/living/carbon/xenomorph/castes/drone/drone.dm [hunk 2]
```diff
@@ -15,3 +15,9 @@
 	inherent_verbs = list(
 		/mob/living/carbon/xenomorph/proc/vent_crawl,
 	)
+
+/mob/living/carbon/xenomorph/drone/on_death()
+	var/datum/action/xeno_action/activable/essence_link/essence_link_action = actions_by_path[/datum/action/xeno_action/activable/essence_link]
+	if(essence_link_action)
+		essence_link_action.end_ability()
+	return ..()
```
Comment (TiviPlus) [inline code/modules/mob/living/carbon/xenomorph/castes/drone/drone.dm:23]: this should be on the ability using signals and not on_death

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by DemonOfHatred** [comment]
love you for fixing this

**Comment by Ragecrabbo** [comment]
hate you for fixing this

File: code/modules/mob/living/carbon/xenomorph/castes/drone/drone.dm (hunk 2 above)
Comment (Penwin0) [reply code/modules/mob/living/carbon/xenomorph/castes/drone/drone.dm:23]: Turns out this is infact redundant, that is already the case

File: code/modules/mob/living/carbon/xenomorph/castes/drone/drone.dm [hunk 3]
```diff
@@ -15,3 +15,4 @@
 	inherent_verbs = list(
 		/mob/living/carbon/xenomorph/proc/vent_crawl,
 	)
+
```
Comment (Penwin0) [inline code/modules/mob/living/carbon/xenomorph/castes/drone/drone.dm:18]:
```suggestion
```

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**